File: YazilimGelistiriciDotNetHomeWork/source/repos/KampHomeWorkJune/Day5Homework5GamerBackEnd/Concretes/GamerManager.cs
using Day5Homework5GamerBackEnd.Abstracts;
using System;
using System.Collections.Generic;
using System.Text;

namespace Day5Homework5GamerBackEnd.Concretes
{
    public class GamerManager:BaseGamerManager
    {
        private IGamerCheckService _gamerCheckService;
        public GamerManager(IGamerCheckService gamerCheckService)
        {
            _gamerCheckService = gamerCheckService;
        }

        public override void Save(Gamer gamer)
        {
            if (_gamerCheckService.CheckIfRealPerson(gamer))
            {
                base.Save(gamer);
            }
            else
            {
                //throw new Exception("Not a valid person");
                Console.WriteLine("Not a valid person");
            }
        }
    }
}

File: JavaReactKampHomeWork/homeworkJavaSE1.8June/src/day4Homework3GamerBackEnd/concretes/GamerManager.java
package day4Homework3GamerBackEnd.concretes;

import day4Homework3GamerBackEnd.abstracts.BaseGamerManager;
import day4Homework3GamerBackEnd.abstracts.GamerCheckService;
import day4Homework3GamerBackEnd.entities.Gamer;

public class GamerManager extends BaseGamerManager{
	private GamerCheckService _gamerCheckService;
	public GamerManager(GamerCheckService gamerCheckService) {
		this._gamerCheckService = gamerCheckService;
		
	}
	public void save(Gamer gamer) {
		
		if(_gamerCheckService.checkIfRealPerson(gamer))
		{
			System.out.println("Gamer Manager : validation successfull");
			super.save(gamer);	
		}else {
			System.out.println("Not a valid person");
		}
		
		
	}

}
